Kernel/Syscalls/chdir.cpp

```diff
@@ -48,11 +48,13 @@ KResultOr<FlatPtr> Process::sys$getcwd(Userspace<char*> buffer, size_t size)
     if (size > NumericLimits<ssize_t>::max())
         return EINVAL;
 
-    auto path = current_directory().absolute_path();
+    auto path = current_directory().try_create_absolute_path();
+    if (!path)
+        return ENOMEM;
 
-    size_t ideal_size = path.length() + 1;
+    size_t ideal_size = path->length() + 1;
```

Comment:
I'd immediately deref this `path` after the `if`, it'd help make things look nicer for one

Reply:
Makes sense, will do.